lib/rubocop/cop/bundler/gem_version_declaration.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Bundler
+      # Enforce that Gem version declarations are either required
+      # or prohibited.
+      #
+      # @example EnforcedStyle: required (default)
+      #  # bad
+      #  gem 'rubocop'
+      #
+      #  # good
+      #  gem 'rubocop', '~> 1.12'
+      #
+      #  # good
+      #  gem 'rubocop', '>= 1.10.0'
+      #
+      #  # good
+      #  gem 'rubocop', '>= 1.5.0', '< 1.10.0'
+      #
+      # @example EnforcedStyle: prohibited
+      #  # good
+      #  gem 'rubocop'
+      #
+      #  # bad
+      #  gem 'rubocop', '~> 1.12'
+      #
+      #  # bad
+      #  gem 'rubocop', '>= 1.10.0'
+      #
+      #  # bad
+      #  gem 'rubocop', '>= 1.5.0', '< 1.10.0'
+      #
+      class GemVersionDeclaration < Base
+        include ConfigurableEnforcedStyle
+
+        REQUIRED_MSG = 'Gem version declaration is required.'
+        PROHIBITED_MSG = 'Gem version declaration is prohibited.'
+        VERSION_DECLARATION_REGEX = /^[~<>=]*\s?[0-9.]+/.freeze
+
+        # @!method gem_declaration?(node)
+        def_node_matcher :gem_declaration?, '(send nil? :gem str ...)'
```

Comment:
This method is in `Bundler::GemComment` as well, it might be useful to extract them to a shared mixin.

Reply:
Yup, definitely. I did copy this from `Bundler::GemComment` to get it working originally. But I completely agree that a mixin will DRY this up.